### PR TITLE
API-127: Move the Java SDK publishing to github actions

### PR DESCRIPTION
**Pull request by Arpit-Sharma-USC**
[JIRA](https://bynder.atlassian.net/browse/API-127)

### CODE REVIEW

**Review by WouterToering — DISMISSED** [review]

File: pom.xml [hunk 1]
```diff
@@ -1,272 +1,345 @@
...
-      <url>https://oss.sonatype.org/content/repositories/snapshots</url>
-    </snapshotRepository>
-    <repository>
-      <id>ossrh</id>
-      <url>https://oss.sonatype.org/service/local/staging/deploy/maven2/</url>
-    </repository>
-  </distributionManagement>
+    <distributionManagement>
+        <!-- Central Repository -->
+        <snapshotRepository>
```
Comment (bankras) [inline pom.xml:21]: It doesn't feel right, that we only specify the snapshot repository

File: pom.xml [hunk 2]
```diff
@@ -1,272 +1,345 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <project xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
-  xmlns="http://maven.apache.org/POM/4.0.0"
-  xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+         xmlns="http://maven.apache.org/POM/4.0.0"
```
Comment (githuib) [inline pom.xml:3]: Any particular reason you changed the indentation from 2 to 4 spaces in this file? Makes it a bit more difficult to see what has changed.

File: pom.xml [hunk 3]
```diff
@@ -1,272 +1,345 @@
...
 
-  <distributionManagement>
-    <snapshotRepository>
-      <id>ossrh</id>
-      <url>https://oss.sonatype.org/content/repositories/snapshots</url>
-    </snapshotRepository>
-    <repository>
-      <id>ossrh</id>
-      <url>https://oss.sonatype.org/service/local/staging/deploy/maven2/</url>
-    </repository>
```
Comment (githuib) [inline pom.xml:27]: I wonder why this was removed?

File: pom.xml [hunk 4]
```diff
@@ -1,272 +1,345 @@
...
+                <groupId>org.sonatype.plugins</groupId>
+                <artifactId>nexus-staging-maven-plugin</artifactId>
+                <version>1.6.8</version>
+                <extensions>true</extensions>
+                <configuration>
+                    <serverId>ossrh</serverId>
+                    <nexusUrl>https://oss.sonatype.org/</nexusUrl>
+                    <autoReleaseAfterClose>true</autoReleaseAfterClose>
+                </configuration>
+            </plugin>
```
Comment (githuib) [inline pom.xml:192]: Shouldn't this be moved to the deployment profile as well?

File: pom.xml (hunk 2 above)
Comment (githuib) [reply pom.xml:3]: Although I almost forgot about github's ignore whitespace toggle :)

File: pom.xml [hunk 5]
```diff
@@ -1,272 +1,345 @@
...
 
-  <build>
-    <resources>
-      <resource>
-        <directory>src/main/resources</directory>
-        <filtering>true</filtering>
-      </resource>
-    </resources>
+    <profiles>
+        <!-- Deployment profile (required so these plugins are only used when deploying) -->
```
Comment (githuib) [inline pom.xml:103]: The 3 plugins below are still listed in the <build><plugins> block as well. Shouldn't they be removed there?

File: pom.xml (hunk 1 above)
Comment (Arpit-Sharma-USC) [reply pom.xml:21]: you mean we should have this and the `repository` as like Huib asked [here](https://github.com/Bynder/bynder-java-sdk/pull/43/files#r549755407) ?

File: pom.xml (hunk 3 above)
Comment (Arpit-Sharma-USC) [reply pom.xml:27]: I think this was overwritten when I was changing the deployment profile. I will add this back.

File: pom.xml (hunk 5 above)
Comment (Arpit-Sharma-USC) [reply pom.xml:103]: Okay I will omit them from there.

File: pom.xml (hunk 2 above)
Comment (Arpit-Sharma-USC) [reply pom.xml:3]: Yeah I just auto-formatted this.

File: pom.xml [hunk 6]
```diff
@@ -1,272 +1,305 @@
...
 
-  <modelVersion>4.0.0</modelVersion>
+    <modelVersion>4.0.0</modelVersion>
 
-  <groupId>com.bynder</groupId>
-  <artifactId>bynder-java-sdk</artifactId>
-  <version>2.2.7</version>
-  <packaging>jar</packaging>
+    <groupId>io.github.arpit-sharma-usc</groupId>
+    <artifactId>byndertest-java-sdk</artifactId>
```
Comment (githuib) [inline pom.xml]: Almost didn't notice this one, because of the indentation changes.. But I think we'll want to restore these two lines ;)

Comment (Arpit-Sharma-USC) [reply pom.xml]: oh no this wasn't there earlier on. I add it test with the test-account I created and then revert it back to bynder Ids. See [here](https://github.com/Bynder/bynder-java-sdk/pull/43/commits/c823f7e33b1b9279c698192a9e6df94d7d8ba86c) 😄

File: pom.xml [hunk 7]
```diff
@@ -1,272 +1,305 @@
...
+    <version>2.2.9</version>
```
Comment (githuib) [inline pom.xml]: I don't think we need a version change for this PR?

Comment (Arpit-Sharma-USC) [reply pom.xml]: No please ignore this as I said I add this to test it with my test-account. (That already has 2.2.8) I original version is still 2.2.7 in the [PR](https://github.com/Bynder/bynder-java-sdk/pull/43/files#diff-9c5fb3d1b7e3b0f54bc5c4182965c4fe1f9023d449017cece3005d3f90e8e4d8R10)

**Review by githuib — APPROVED** [review]

**Pull request merged**